src/sentry/event_manager.py

```diff
@@ -142,7 +143,18 @@ def has_pending_commit_resolution(group):
     if recent_group_link is None:
         return False
 
-    return not ReleaseCommit.objects.filter(commit__id=recent_group_link.linked_id).exists()
+    # commit has been released and is not in pending commit state
+    if ReleaseCommit.objects.filter(commit__id=recent_group_link.linked_id).exists():
+        return False
+    else:
+        # check if this commit is a part of a PR
+        if PullRequestCommit.objects.filter(commit__id=recent_group_link.linked_id).exists():
+            # check if any commits in the PR have been released
+            if ReleaseCommit.objects.filter(
+                commit__pullrequest_commit__commit_id=recent_group_link.linked_id
+            ).exists():
```

Comment:
@wedamija I’m getting a FieldError with this query - I’m trying to get the commits that are a part of the PR that have been released. Is there a way to filter on the pr's id instead of the commit id?

Reply:
One way to figure out the correct relation names here is to jump into `sentry shell` and play around. You can do something like
```
from sentry.models import *

ReleaseCommit.objects.filter(a=1)
<trimmed error message>
FieldError: Cannot resolve keyword 'a' into field. Choices are: commit, commit_id, id, order, organization_id, project_id, release, release_id

Commit.objects.filter(a=1)
<trimmed error message>
FieldError: Cannot resolve keyword 'a' into field. Choices are: author, author_id, commitfilechange, date_added, id, key, message, organization_id, pullrequestcommit, releasecommit, releaseheadcommit, repository_id
```

From this you can get the full path, in this case `ReleaseCommit.objects.filter(commit__pullrequestcommit__commit_id=1)`

Reply:
@wedamija Thanks for the tip!